### PR TITLE
HHH-14524 : Rename SqlType(X) as JdbcType(X)

### PR DESCRIPTION
**Pull request by sebersole**

### CODE REVIEW

File: documentation/src/main/asciidoc/userguide/chapters/domain/jdbc_mappings.adoc [hunk 1]
```diff
@@ -29,13 +29,13 @@ An `AttributeConverter` can also influence the `JavaTypeDescriptor` used in cert
 
 ==== SqlTypeDescriptor
 
-`@SqlType` can be used to explicitly specify the `SqlTypeDescriptor` to use for a particular mapping.
+`JdbcType` can be used to explicitly specify the `JdbcTypeDescriptor` to use for a particular mapping.
 
-`@SqlTypeCode` can be used to indicate the `SqlTypeDescriptorRegistry` entry to use.  `@SqlTypeRegistration` can
-be used to register a `SqlTypeDescriptor` with the `SqlTypeDescriptorRegistry`.
+`JdbcTypeCode` can be used to indicate the `JdbcTypeDescriptorRegistry` entry to use.  `JdbcTypeRegistration` can
```
Comment (beikov) [inline documentation/src/main/asciidoc/userguide/chapters/domain/jdbc_mappings.adoc:34]:
```suggestion
`@JdbcTypeCode` can be used to indicate the `JdbcTypeDescriptorRegistry` entry to use.  `@JdbcTypeRegistration` can
```

File: documentation/src/main/asciidoc/userguide/chapters/domain/jdbc_mappings.adoc [hunk 2]
```diff
@@ -29,13 +29,13 @@ An `AttributeConverter` can also influence the `JavaTypeDescriptor` used in cert
...
+be used to register a `JdbcTypeDescriptor` with the `JdbcTypeDescriptorRegistry`.
 
-`@MapKeySqlType` and `@MapKeySqlTypeCode` work the same as `@SqlType` and `@SqlTypeCode`, respectively, except
-describing the `SqlTypeDescriptor` to use for the map's key.  In which case, `@SqlType` and `@SqlTypeCode` refer
+`MapKeyJdbcType` and `MapKeyJdbcTypeCode` work the same as `JdbcType` and `JdbcTypeCode`, respectively, except
```
Comment (beikov) [inline documentation/src/main/asciidoc/userguide/chapters/domain/jdbc_mappings.adoc:37]:
```suggestion
`@MapKeyJdbcType` and `@MapKeyJdbcTypeCode` work the same as `@JdbcType` and `@JdbcTypeCode`, respectively, except
```

File: documentation/src/main/asciidoc/userguide/chapters/domain/jdbc_mappings.adoc [hunk 3]
```diff
@@ -29,13 +29,13 @@ An `AttributeConverter` can also influence the `JavaTypeDescriptor` used in cert
...
+describing the `JdbcTypeDescriptor` to use for the map's key.  In which case, `JdbcType` and `JdbcTypeCode` refer
```
Comment (beikov) [inline documentation/src/main/asciidoc/userguide/chapters/domain/jdbc_mappings.adoc:38]:
```suggestion
describing the `JdbcTypeDescriptor` to use for the map's key.  In which case, `@JdbcType` and `@JdbcTypeCode` refer
```

**Review by beikov — COMMENTED** [review]
Looks good. I just found some small documentation changes that seem unintended.

**Comment by sebersole** [comment]
Applied upstream

**Pull request closed without merging**